examples/04_New Models.ipynb

```diff
@@ -993,6 +993,173 @@
+   "outputs": [],
+   "source": [
+    "event = 'impact'\n",
+    "simulated_results = m_limits.simulate_to_threshold(threshold_keys=[event], dt=0.005, save_freq=1)\n",
+    "\n",
+    "print('Example: With State Limits')\n",
+    "for i, state in enumerate(simulated_results.states):\n",
+    "    print(f'State {i}: {state}')\n",
+    "print()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that now the position (`x`) becomes 0 but never reaches a negative value. This is because we have defined a state limit for the `x` state that prevents it from going below 0. Also note that a warning is provided to notify the user that a state value was limited. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's try a more complicated example. This time, we'll try setting the initial position value to be a number outside of its bounds. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x0 = m_limits.initialize(u = {}, z = {})\n",
+    "x0['x'] = -1 # Initial position value set to an unrealistic value of -1\n",
+    "\n",
+    "simulated_results = m_limits.simulate_to_threshold(threshold_keys=[event], dt=0.005, save_freq=1, x = x0)\n",
+    "\n",
+    "# Print states\n",
+    "print('Example 2: With -1 as initial x value')\n",
+    "for i, state in enumerate(simulated_results.states):\n",
+    "    print('State ', i, ': ', state)\n",
+    "print()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that the simulation stops after just two iterations. In this case, the initial position value is outside the state limit. On the first iteration, the position value is therefore adjusted to be within the appropriate range of 0 to $\\infty$. Since we are simulating to impact, which is defined as when position is 0, the threshold is immediately satisfied and the simulation stops. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, note that limits can also be applied manually using the `apply_limits` function. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x = {'x': -5, 'v': 3e8}  # Too fast and below the ground\n",
+    "print('\\t Pre-limit: {}'.format(x))\n",
+    "\n",
+    "x = m_limits.apply_limits(x)\n",
+    "print('\\t Post-limit: {}'.format(x))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In conclusion, setting appropriate [state limits](https://nasa.github.io/progpy/prog_models_guide.html#state-limits)  is crucial in creating realistic and accurate state-transition models. It ensures that the model's behavior stays within the constraints of the physical system. The limits should be set based on the physical or practical constraints of the system being modeled. "
```

Comment:
Somewhere I want to also add that state limits are especially important for state estimation, as it will force the state estimator to only consider possible states. Let's also reference the state estimation section